R/utils.R

```diff
@@ -448,6 +448,118 @@ guess_time_type <- function(time_value) {
   return("custom")
 }
 
+#'  given a vector of characters, add the same values, but upcased, e.g.
+#'   "date" -> c("date", "Date")
+#'   "target_date" -> c("target_date", "Target_Date")
+#' @keywords internal
+upcase_snake_case <- function(vec) {
+  upper_vec <- strsplit(vec, "_") %>%
+    map(function(name) paste0(toupper(substr(name, 1, 1)), substr(name, 2, nchar(name)), collapse = "_")) %>%
+    unlist()
+  c(vec, upper_vec)
+}
+
+#' rename potential time_value columns
+#' @keywords internal
+guess_time_column_name <- function(x, substitutions = NULL) {
+  if (!("time_value" %in% names(x))) {
+    if (is.null(substitutions)) {
+      substitutions <- c(
+        time_value = "date",
+        time_value = "time",
```

Comment:
suggestion (optional): reduce typing for the future by setting the names all at once, like

```
substitutions <- c("date", "time", ...)
names(substitutions) <- rep("time_value", length(substitutions))
```

Reply:
and for geo and version fns